### PR TITLE
rework workers logic completly

### PR DESCRIPTION
**Pull request by coolo**
We noticed that the new worker did not appear on admin/worker page and
I couldn't resist doing it properly

### CODE REVIEW

File: lib/OpenQA/Controller/Admin/Workers.pm [hunk 1]
```diff
@@ -17,28 +17,21 @@
...
+    my $workers = $self->db->resultset('Workers');
+    $self->stash(workers => $workers);
 
     $self->render('admin/workers/index');
 }
 
 sub show {
-    my $self = shift;
-    my $workerid = $self->param('worker_id');
-    $self->stash('id', $workerid);
```
Comment (aaannz) [inline lib/OpenQA/Controller/Admin/Workers.pm]: $id still used at template admin/workers/show.html.ep:2

File: templates/admin/workers/index.html.ep [hunk 2]
```diff
@@ -2,62 +2,68 @@
...
+	<tbody>
+	    % my %workers; # for sorting
+	    % while (my $w = $workers->next) {
+		% next unless $w->id;
+		% my $workername = $w->host . ":" . $w->instance;
+		% $workers{$workername} = $w;
+	    % }
+	    % for my $workername (sort keys %workers) {
+		%   my $view = $workers{$workername};
+		<tr id="worker_<%= $view->id %>" >
```
Comment (aaannz) [inline templates/admin/workers/index.html.ep]: The same you did here (+20lines) needs to be done to admin/workers/show.html.ep too (from hash to method calls).

**Comment by coolo** [comment]
I thought I catch you to review this with some nice summary :)

File: lib/OpenQA/Controller/Admin/Workers.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/Controller/Admin/Workers.pm]: I guess we lack even more test coverage :(

**Comment by coolo** [comment]
done

**Pull request merged**